spec/services/tag_importer/links_publisher_spec.rb

```diff
@@ -0,0 +1,153 @@
+require 'rails_helper'
+
+RSpec.describe TagImporter::LinksPublisher do
```

Comment:
I find this test quite hard to follow. Could we make it smaller and do less, or move parts to an integration test? Feels like it's trying to do too much.